res/controllers/engine-api.d.ts

```diff
@@ -72,6 +73,28 @@ declare namespace engine {
      */
     function setParameter(group: string, name: string, newValue: number): void;
 
+    /**
+     * Gets the shared runtime data.
+     * @returns Runtime shared data value
+     */
+    function getSharedData(): SharedDataValue;
```

Comment:
```suggestion
    /**
     * Gets the shared runtime data.
     * @returns Runtime shared data value - or undefined if not yet defined
     */
    function getSharedVariable(variablename:string): SharedDataValue;
```
We should not share one single complex blob. This makes it really difficult to understand for developers and we can't print meaningful error messages, when two mappings do no match together.
The later can especially happen, when you edit the variable type at runtime of Mixxx, because only one mapping is reloaded at a time.

Reply:
> We should not share one single complex blob.

What if, in your namespace `foobar`, you want to share just a `TypedArray` or a string, like in the tests? I think adding variables don't add any values to the feature and just another level.

> we can't print meaningful error messages, when two mappings do no match together.

This isn't possible anyway. I thought we agreed that introducing namespace was a good enough option to prevent conflict. 
I also don't see how this would work anyway either since any controllers part of a namespace can get/edit values as well: this means that it is acceptable for `MappingA` to do `get("MappingBRunning")` (or `get().MappingBRunning` currently) and find no values.

Reply:
Namespacing is good enough to prevent unintended iterference of completly unrelated mappings.
But this does not address compatibility between the related scripts using the same namespace.

Reply:
I think this request is consistent with my original namespacing proposal above: https://github.com/mixxxdj/mixxx/pull/12199#issuecomment-2038000562

Reply:
How your solution will compatibility between the related scripts compare to the current one? Do you mean I should also implement the [API change mentioned here](https://github.com/mixxxdj/mixxx/pull/12199#issuecomment-2106768820)?

Reply:
This would be one approach. But also without an explicit declaration, it would be possible to print warnings like:
`Line 1234 - Shared variable "deckSwitch" changed type in call of engine.setSharedVariable from number to string`

Reply:
This is all javascript, there is no schema / static typing, nor is it really necessary. If there is no technical reason why variables could not change their type I don't see a good reason to introduce one (especially when it would only result in extra complexity).

Reply:
While I understand you are using type change as a way to detect a potential problem, this lefts out other usecase where a type change wouldn't occur but yet doesn't occur - eg wrong unit.
I think it all boils down to understanding what you are sharing, and making sure producers and consumers expect the same data AND don't conflict with each other. 
I think this type of detection is a nice to have, but I cannot really see how this is required. Realistically, only a hand full of official mappings will use this features, so I think it is a matter of making sure this is well document (e.g using shared context class for example).
If you really feel like there should be a key for data set/get, I will add it, but lets agree first this is the last significant change I'm being asked to do.

Reply:
for the record, I don't think a schema a necessary. exchanging an untyped/dynamic object is completely fine.

Reply:
@JoergAtGithub is that a hard requirement in your opinion, or are you happy to go with the untyped dynamic data as it is currently?